### PR TITLE
jaguar vector store

### PR DESCRIPTION
**Pull request by fserv**
# Description

This PR is adding a new vector store that will work with LLamaIndex. The new vector store is a distributed database system that can store a large amount of vectors as well as scalars on multiple server hosts. With growing number of vectors, the system can scale horizontally and instantly to accommodate more vectors, with its ZeroMove sharding framework. The jaguar distributed vector store also supports anomaly detection which can help identify outlier data. With the time-cutoff and time-decay modulation mechanism, similarity score of history data is adjusted according to their creation time, for applications where timestamp of events or descriptions plays an important role.

Dependency:
    pip install -U jaguardb-http-client

## Type of Change

- [ x] New feature:  jaguar vector store 

# How Has This Been Tested?

Firstly, the jaguar vector store server software is installed on one or more computer hosts. Secondly, the http gateway server for jaguar vector store server is installed. The steps are described http://www.jaguardb.com  and related documentation on the web site.  Installation of jaguar server can be done with docker or non-docker environments (any Linux). Installation of http gateway server can be done on any Linux systems (platform independent).

- [x ] Added new unit/integration tests
- [x ] Added new notebook (that tests end-to-end)
- [x ] I stared at the code and made sure it makes sense

# Checklist:

- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] I have added Google Colab support for the newly added notebooks.
- [x ] My changes generate no new warnings
- [x ] I have added tests that prove that my feature works
- [x ] New unit tests pass locally with my changes
- [x ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by jonyue — APPROVED** [review]

**Comment by hatianzhang** [comment]
this looks awesome with new Jaguar vector db. Could @fserv help fix some units tests first? you could mock vector db client for unit tests.
Also since this is a new vector db, you better to add some documentation for this new vdb.
like `/llama_index/docs/community/integrations/vector_stores.md`

**Comment by fserv** [comment]
> this looks awesome with new Jaguar vector db. Could @fserv help fix some units tests first? you could mock vector db client for unit tests. Also since this is a new vector db, you better to add some documentation for this new vdb. like `/llama_index/docs/community/integrations/vector_stores.md`

Thanks @hatianzhang very much for the suggestions. The changes are made. Could you please review again?

File: docs/examples/vector_stores/JaguarIndexDemo.ipynb [hunk 1]
```diff
@@ -0,0 +1,148 @@
+{
```
Comment (hatianzhang) [inline docs/examples/vector_stores/JaguarIndexDemo.ipynb:1]: <p>the format of each step are not well-formated</p><p> </p><pre class="ql-syntax" spellcheck="false">1) It is a distributed vector database that can store large number of vectors. &nbsp;2) The ZeroMove feature of JaguarDB enables instant horizontal scalability. &nbsp;3) It works with multimodal data including vector embeddings, text, images, videos, PDFs, audio, time series, and geospatial data.  &nbsp;4) The all-master architecture allows both parallel reads and writes. &nbsp;5) Its anomaly detection capabilities can distinguish outliers in the dataset. &nbsp;6) The RAG support can combine LLMs and proprietary and real-time data. &nbsp;7) Sharing of metadata across multiple vector indexes improves data consistency. &nbsp;8) Distance metrics include Euclidean, Cosine, InnerProduct, Manhatten, Chebyshev, Hamming, Jeccard, and Minkowski. &nbsp;9) Similarity search can be performed with time cutoff and time decay effects. 



</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/9754/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (hatianzhang) [inline docs/examples/vector_stores/JaguarIndexDemo.ipynb:1]: <p>could you break down this into different sections and add outputs for each step</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/9754/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Comment by fserv** [comment]
@hatianzhang Could you review please at your convenience? Thanks much.

**Review by hatianzhang — APPROVED** [review]
awesome! LGTM!

**Comment by fserv** [comment]
> awesome! LGTM!

Appreciate your time and suggestions/reviews, especially during holidays.

**Pull request merged**